mfeepo_connector.py

```diff
@@ -212,58 +232,52 @@ def __add_tag(self, param, quarantine=False):
         if quarantine:
             tag = self.get_config().get(EPO_JSON_QTAG, "")
             if not tag:
-                self.set_status(phantom.APP_ERROR)
                 return action_result.set_status(phantom.APP_ERROR, "Please provide the quarantine tag in asset configuration parameter")
         else:
             tag = param[EPO_JSON_TAG]
 
-        action_result, tag = self._validate_tag(action_result, host, tag)
+        ret_val, tag = self._validate_tag(action_result, host, tag)
 
         # Tag doesn't exist
-        if phantom.is_fail(action_result.get_status()):
+        if phantom.is_fail(ret_val):
             return action_result.get_status()
 
-        action_result, r_dict = self._find(action_result, host)
+        ret_val, r_dict = self._find(action_result, host)
 
         # Host doesn't exist
-        if phantom.is_fail(action_result.get_status()):
+        if phantom.is_fail(ret_val):
             return action_result.get_status()
 
         # Endpoint already has tag
         if self._check_tag(r_dict[KEY_TAGS], tag):
             return action_result.set_status(phantom.APP_SUCCESS, "Success, tag already added")
 
-        action_result = self._apply_tag(action_result, host, tag)
+        ret_val = self._apply_tag(action_result, host, tag)
 
-        if not wakeup_agent or phantom.is_fail(action_result.get_status()):
+        if not wakeup_agent or phantom.is_fail(ret_val):
             return action_result.get_status()
 
         ret_val = self._wakeup_agent(action_result, host)
 
         if phantom.is_fail(ret_val):
             return action_result.set_status(phantom.APP_ERROR,
-                    "Assigned tag but host did not recieve configuration")
+                                            "Assigned tag but host did not receive configuration")
 
         return action_result.set_status(phantom.APP_SUCCESS,
-                "Assigned tag and host received configuration")
+                                        "Assigned tag and host received configuration")
 
     def _apply_tag(self, action_result, host, tag):
 
         try:
             ret_val, _ = self._make_rest_call('system.applyTag', {'param1': host, 'param2': tag}, action_result)
 
-        except:  # Something went wrong
-            self.set_status(phantom.APP_ERROR)
-            action_result.set_status(phantom.APP_ERROR, "Failed to assign tag")
-            return action_result
+        except Exception:  # Something went wrong
+            return action_result.set_status(phantom.APP_ERROR, "Failed to assign tag")
 
         if ret_val == 0:  # Something else went wrong
```

Comment:
The _make_rest_call method returns the boolean value in the first parameter.
We are comparing a boolean with an integer value. We can use `phantom.is_fail(ret_val)` instead.

Reply:
As the code was using the response value earlier, instead of checking the ret_val value will now use the response value which is an integer.

Reply:
Okay resolving the thread.